lib/browser/storage.js

```diff
@@ -22,17 +22,25 @@ try {
 
 export const canStoreURLs = hasStorage;
 
-export function setItem(key, value) {
-  if (!hasStorage) return;
-  return localStorage.setItem(key, value);
-}
+export class DefaultStorage {
+  setItem (key, value) {
+    return new Promise((resolve, reject) => {
```

Comment:
Sure! I think I misunderstood your comment here:

> Promises as this is probably going to be the standard in new browser APIs

That's why I went with the Promise approach. Will make changes 👍